Library/Homebrew/cmd/update.sh

```diff
@@ -428,8 +428,12 @@ EOS
     echo "HOMEBREW_BREW_GIT_REMOTE set: using $HOMEBREW_BREW_GIT_REMOTE for Homebrew/brew Git remote."
     git remote set-url origin "$HOMEBREW_BREW_GIT_REMOTE"
     git config remote.origin.fetch "+refs/heads/*:refs/remotes/origin/*"
-    latest_tag="$(git ls-remote --tags --refs --sort=v:refname -q origin | tail -n1 | cut -f2)"
-    git fetch --force origin --shallow-since="$latest_tag"
+    latest_tag="$(git ls-remote --tags --refs -q origin |
+                  cut -d/ -f3 | 
+                  sort --field-separator=. --key=1,1nr -k 2,2nr -k 3,3nr|
+                  head -n1)"
```

Comment:
Thanks, @shcai97.

Let's be consistent and use `--key` flag instead of `-k` for the second and third fields too.

Field-specific ordering options (`nr`) are used to override "global" ordering options. I wonder if it would be more logical to set global ordering using `--numeric-sort` and don't override field-specific ones (@MikeMcQuaid):

```suggestion
                  sort --numeric-sort --field-separator=. --key=1,1 --key=2,2 --key=3,3 |
                  tail -n1)"
```

I also don't see the need to reverse the sorting order -- let's remove those `r`s and use `tail` instead of `head` -- I did that in the snippet above.